main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -80,17 +82,27 @@ object Namer extends Phase[WeededAst.Program, NamedAst.Root] {
         visitDecl(decl, namespace, uenv0, pacc)
     }
 
-    case decl@WeededAst.Declaration.Class(doc, mod, ident, tparam, signatures, loc) =>
-
+    case decl@WeededAst.Declaration.Class(doc, mod, ident, tparam, sigs, loc) =>
+      // Check if the class already exists.
       val classes = prog0.classes.getOrElse(ns0, Map.empty)
       classes.get(ident.name) match {
         case None =>
+          // Case 1: The class does not already exist. Update it.
           visitClass(decl, uenv0, Map.empty, ns0) map {
             clazz => prog0.copy(classes = prog0.classes + (ns0 -> (classes + (ident.name -> clazz))))
           }
         case Some(clazz) =>
+          // Case 2: Duplicate class.
           NameError.DuplicateClass(ident.name, clazz.sym.loc, ident.loc).toFailure
       }
+
+    case decl@WeededAst.Declaration.Instance(doc, mod, clazz, tpe0, defs, loc) =>
+      // duplication check must come after name resolution
```

Comment:
Yes, especially because it possible to have overlapping instances, e.g. say `Eq(a, a)` and `Eq(Int, a)`.